### PR TITLE
algorithms_task_1_kostiuchenko

### PR DESCRIPTION
**Pull request by kosnko**

### CODE REVIEW

File: algorithms_task_1/student/include/impl.h [hunk 1]
```diff
@@ -11,4 +11,9 @@
   * @return consecutive range
 */
 template<class FI, class Comparator = std::equal_to<typename std::iterator_traits<FI>::value_type>>
-std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {});
+std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {})
+{
+    FI it1 = std::adjacent_find(start, end, comp);
+    FI it2 = (it1 != end) ? std::find_if_not( std::next(it1), end, [&](typename std::iterator_traits<FI>::value_type elem){return comp(elem, *it1);} ) : end;
```
Comment (smay1613) [inline algorithms_task_1/student/include/impl.h:17]: Don't copy to lambda

Comment (smay1613) [reply algorithms_task_1/student/include/impl.h:17]: Please separate this expression, because currently it is unreadable

Comment (kosnko) [reply algorithms_task_1/student/include/impl.h]: Done

**Review by kosnko — COMMENTED** [review]
Updated

File: algorithms_task_1/student/include/impl.h [hunk 2]
```diff
@@ -14,6 +14,12 @@ template<class FI, class Comparator = std::equal_to<typename std::iterator_trait
 std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {})
 {
     FI it1 = std::adjacent_find(start, end, comp);
-    FI it2 = (it1 != end) ? std::find_if_not( std::next(it1), end, [&](typename std::iterator_traits<FI>::value_type elem){return comp(elem, *it1);} ) : end;
+    FI it2 = (it1 != end) ? std::find_if_not( std::next(it1), end,
```
Comment (smay1613) [inline algorithms_task_1/student/include/impl.h:17]: prefer if statement for such cases

**Pull request closed without merging**